iModelCore/ECPresentation/Tests/NonPublished/Integration/Hierarchies/HierarchyUpdateTests.cpp

```diff
@@ -10,4906 +10,1413 @@ USING_NAMESPACE_BENTLEY_SQLITE_EC
+    </ECEntityClass>
+)*");
+TEST_F(HierarchyUpdateTests, GroupByLabel_DetectsGroupedNodeLabelChanged)
     {
+    // set up dataset
+    ECClassCP classA = GetClass("A");
+    IECInstancePtr a11 = RulesEngineTestHelpers::InsertInstance(m_db, *classA, [](IECInstanceR instance){instance.SetValue("LabelProp", ECValue("1")); });
+    IECInstancePtr a12 = RulesEngineTestHelpers::InsertInstance(m_db, *classA, [](IECInstanceR instance){instance.SetValue("LabelProp", ECValue("1")); });
+    IECInstancePtr a21 = RulesEngineTestHelpers::InsertInstance(m_db, *classA, [](IECInstanceR instance){instance.SetValue("LabelProp", ECValue("2")); }, true);
+
     // create the rule set
     PresentationRuleSetPtr rules = PresentationRuleSet::CreateInstance(BeTest::GetNameOfCurrentTest());
     m_locater->AddRuleSet(*rules);
 
-    rules->AddPresentationRule(*new InstanceLabelOverride(1, true, "RulesEngineTest:Widget", "MyID"));
-    RootNodeRule* rule = new RootNodeRule("", 1, false, RuleTargetTree::TargetTree_Both, false);
-    rule->AddSpecification(*new InstanceNodesOfSpecificClassesSpecification(1, false, false, false, false, false, false, "", "RulesEngineTest:Gadget", false));
-    rule->AddSpecification(*new InstanceNodesOfSpecificClassesSpecification(1, false, false, false, false, false, false, "", "RulesEngineTest:Widget", false));
-    rules->AddPresentationRule(*rule);
-
-    // request for root nodes
-    DataContainer<NavNodeCPtr> rootNodes = RulesEngineTestHelpers::GetValidatedNodes([&](){ return m_manager->GetNodes(AsyncHierarchyRequestParams::Create(m_db, rules->GetRuleSetId(), RulesetVariables())).get(); });
-
-    // expect no nodes
-    ASSERT_EQ(0, rootNodes.GetSize());
+    RootNodeRule* rootRule = new RootNodeRule("", 1, false, RuleTargetTree::TargetTree_Both, false);
+    rootRule->AddSpecification(*new InstanceNodesOfSpecificClassesSpecification(1, ChildrenHint::Unknown, false, false, false, true, "",
+        {
+        new MultiSchemaClass(classA->GetSchema().GetName(), true, bvector<Utf8String>{ classA->GetName() })
+        }, {}));
+    rules->AddPresentationRule(*rootRule);
+    rules->AddPresentationRule(*new InstanceLabelOverride(1, false, classA->GetFullName(), { new InstanceLabelOverridePropertyValueSpecification("LabelProp") }));
 
-    // create a widget
-    IECInstancePtr widget = RulesEngineTestHelpers::InsertInstance(m_db, *m_widgetClass, [](IECInstanceR instance){instance.SetValue("MyID", ECValue("WidgetID"));}, true);
-    m_eventsSource->NotifyECInstanceInserted(m_db, *widget);
+    // validate hierarchy pre-update
+    auto params = AsyncHierarchyRequestParams::Create(m_db, rules->GetRuleSetId(), RulesetVariables());
+    ValidateHierarchy(params,
+        {
+        ExpectedHierarchyDef(CreateLabelGroupingNodeValidator("1", { a11, a12 }),
+            {
+            CreateInstanceNodeValidator({ a11 }),
+            CreateInstanceNodeValidator({ a12 }),
+            }),
+        CreateInstanceNodeValidator({ a21 }),

```

Comment:
maybe it's also worth having a test case for removing the sibling node (in this case `a21`)

Reply:
Added test to check if LabelGrouping node is removed when it is only one left at the root